expression/scalar_function.go

```diff
@@ -39,6 +39,24 @@ type ScalarFunction struct {
 	RetType  *types.FieldType
 	Function builtinFunc
 	hashcode []byte
+
+	// fields for vectorized evaluation
+	testVectorized bool
+	vectorized     bool
+}
+
+// Vectorized returns if this expression supports vectorized evaluation.
+func (sf *ScalarFunction) Vectorized() bool {
+	if !sf.testVectorized {
+		sf.vectorized = sf.Function.vectorized()
+		sf.testVectorized = true
```

Comment:
Is it possible that this has data race?

Reply:
Why not call `sf.Function.vectorized` directly?

Reply:
The `sf.Function.vectorized()` will check the whole expression tree.
In row mode, if we do it before evaluating each row, the performance will be badly affected.

Reply:
How about replacing it with an atomic variable.

Reply:
Can we set `vectorized` attribute when creating the function?

Reply:
Yes, good idea.

Reply:
Fixed.